File: src/main/java/com/smhrd/domain/Board.java
package com.smhrd.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

@Data //@Data이거 치면 기본 매서드 다 만들어줌 굳
@AllArgsConstructor // 모든 필드 생성자
@NoArgsConstructor // 기본 생성자
@RequiredArgsConstructor //필요한 요소만 가진 생성자
public class Board {

	// 필드명 == 컬럼명
	// 기본생성자는 반드시 있어야함
	
	
	// 글번호, 제목, 작성자, 내용, 작성일, 조회수
	
	private int idx;
	@NonNull
	private String title;
	@NonNull
	private String writer;
	@NonNull
	private String content;
	private String indate;
	private int count;

}

File: src/main/webapp/resources/js/board.js
$(document).ready( (e) => {
	$.ajax({
		type :"post",
		url : "boardList.do",
		data : {
			
		},
		dataType : "json",
		success : function(data){
			
			let table ="";
			
			for(let i=0;i<data.length;i++){
			// thyemleaf : html에서 쓰는 jstl
				table += `
				<tr><td>${data[i].idx}</td>
				<td><a href="javascript:open(${data[i].idx},${data[i].count})">${data[i].title}</a></td>
				<td>`+data[i].writer+`</td>
				<td>`+data[i].indate+`</td>
				<td>`+data[i].count+`</td></tr>
				<tr id="board${data[i].idx}" style="display:none">
				<td colspan = 5>
				<textarea row="5" class="form-control">${data[i].content}</textarea>
				</td>
				</tr>
				`
				
			}
			$('#tbody').html(table);
			
			
		},
		error : function(e){
			alert('error')
		}
	})
})
//====================================================================================

function open(idx, count){

	if($('#board'+idx).css('display') =='none'){
		
		$('#board'+idx).css('display','table-row');
		
		$.ajax({
			type :"get",
			url : "viewUpdate.do",
			data : {
				"idx":idx,
				"count":count
			},
			dataType : "text",
			success : function(data){
			},
			error : function(e){
				alert('view error')
			}
		})

	}else{
		$('#board'+idx).css('display','none')
	}

}